include/alpaka/exec/ExecCpuTbbBlocks.hpp

```diff
@@ -0,0 +1,324 @@
+
+                // Get the size of the block shared dynamic memory.
+                auto const blockSharedMemDynSizeBytes(
+                    meta::apply(
+                        [&](TArgs const & ... args)
+                        {
+                            return
+                                kernel::getBlockSharedMemDynSizeBytes<
+                                    acc::AccCpuTbbBlocks<TDim, TSize>>(
+                                        m_kernelFnObj,
+                                        blockThreadExtent,
+                                        threadElemExtent,
+                                        args...);
+                        },
+                        m_args));
+
+#if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
+                std::cout << BOOST_CURRENT_FUNCTION
+                    << " blockSharedMemDynSizeBytes: " << blockSharedMemDynSizeBytes << " B" << std::endl;
+#endif
+                // Bind all arguments except the accelerator.
+                // TODO: With C++14 we could create a perfectly argument forwarding function object within the constructor.
+                auto const boundKernelFnObj(
+                    meta::apply(
+                        [this](TArgs const & ... args)
+                        {
+                            return
+                                std::bind(
+                                    std::ref(m_kernelFnObj),
+                                    std::placeholders::_1,
+                                    std::ref(args)...);
+                        },
+                        m_args));
+
+
+
+                // The number of threads in this block.
+                //TSize const blockThreadCount(blockThreadExtent.prod());
+
+                // The number of blocks in the grid.
+                TSize const numBlocksInGrid(gridBlockExtent.prod());
+
+                // There is only ever one thread in a block in the TBB accelerator.
+                assert(blockThreadExtent.prod() == 1u);
+
+                tbb::task_group g;
+
+                for(TSize i = 0; i < numBlocksInGrid; ++i)
+                {
+                         g.run([&]{
+
+                             acc::AccCpuTbbBlocks<TDim, TSize> acc(
+                                 *static_cast<workdiv::WorkDivMembers<TDim, TSize> const *>(this),
+                                 blockSharedMemDynSizeBytes);
+
+                             acc.m_gridBlockIdx =
+                                 core::mapIdx<TDim::value>(
+                                     Vec<dim::DimInt<1u>, TSize>(
+                                         static_cast<TSize>(i)
+                                      ),
+                                      gridBlockExtent
+                                 );
+
+                             boundKernelFnObj(acc);
+
+                             block::shared::st::freeMem(acc);
+
+                         });
+
+                         g.wait();
```

Comment:
Is it correct to wait here? Shouldn`t it be outside the loop to execute all blocks in parallel and wait for all of them to be finished?